x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/tag_options.tsx

```diff
@@ -78,29 +108,50 @@ export const TagOptions: React.FC<Props> = ({ tagName, isTagHovered, onTagsUpdat
         >
           <EuiFlexGroup direction="column" alignItems="flexStart" gutterSize="xs">
             <EuiFlexItem>
-              <EuiFieldText
-                placeholder={i18n.translate('xpack.fleet.tagOptions.nameTextFieldPlaceholder', {
-                  defaultMessage: 'Enter new name for tag',
-                })}
-                value={updatedName}
-                required
-                onChange={(e: any) => {
-                  const newName = e.target.value;
-                  setUpdatedName(newName);
-                  if (!newName) {
-                    return;
-                  }
-                  debouncedSendRenameTag(newName);
-                }}
-              />
+              <EuiFlexGroup gutterSize="xs">
+                <EuiFlexItem>
+                  <EuiFieldText
+                    placeholder={i18n.translate('xpack.fleet.tagOptions.nameTextFieldPlaceholder', {
+                      defaultMessage: 'Enter new name for tag',
+                    })}
+                    value={updatedName}
+                    required
+                    onKeyDown={(e: { key: string }) => {
+                      if (e.key === 'Enter') {
+                        handleRename(updatedName!);
+                        closePopover();
+                      }
+                    }}
+                    onChange={(e: any) => {
+                      const newName = e.target.value;
+                      setUpdatedName(newName);
+                    }}
+                  />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <EuiButtonIcon
+                    size="m"
+                    display="fill"
+                    iconType="save"
```

Comment:
The save icon seems a little out of place in my (subjective and poorly informed) design opinion. Maybe `check` would be better?  @dborodyansky might have input, but let's not block a merge on it if he does.

Reply:
okay, I'm also okay to remove the button completely since we have the other 2 ways to save and that was the intention in the design

Reply:
Let's remove it then. Thanks!

Reply:
done